acp/identity/identity.go

```diff
@@ -17,6 +17,15 @@ import (
 	"github.com/sourcenetwork/immutable"
 )
 
+// didProducer is a concrete function which
+// produces a did from a given key type and pub key bytes
+//
+// Currently the did production operation is
+// infalliable, but in order to assure the correct error
+// is being returned, this pkg private variable
+// can be used in tests to locally mock the producer function
+var didProducer = key.CreateDIDKey
```

Comment:
Very good catch, thanks Shahzad :)

Reply:
I really don't like this suggestion. I don't think callers of a package should be required to inject the concrete function implementation.

The package is succint enough such that there's no confusion about what the didProducer  means, furthermore it is simple to the point that we get the desired coverage / protection against changes breaking changes in the upstream libraries without adding abstraction for the sake of abstractions.

I'll change this construction but I *really* don't think it's worth it. Adding 1/2 interfaces to achieve essentially what will be an elaborate mock of 1 function (which doesn't have an alternative implementation) in a package with a single file screams unnecessary complexity

Reply:
I had another look, @Lodek why is the test mutating this?  Why are you not just calling `key.CreateDIDKey` in `DIDFromPublicKey`?

Reply:
> Is it just to get code coverage over a dead-code line

Pretty much. Shahzad mentioned he had a preference to having coverage over that test scenario. I ended redesigning things a bit as to avoid the global mutation and still getting the coverage that was requested.

Reply:
todo: You got rid of the global (I think), but the cure still looks way worse than the disease. @shahzadlone can we please not do this? (sorry Bruno for you being dragged back and forth on this)

Reply:
I don't necessarily care about testing for this scenario, if it will make us over abstract things.

I am not sure why the interface needed to be introduced, I agree with Andy the new changes aren't better than what we had previously.

My main concern was the previous glabal var.

If you don't want to let package inject / pass it. Then why not simply keep the previous implementation and just set that variable locally within that function.

The prev implementation was good, just needed to get rid of glabal var.

Reply:
I reverted the changes. Now there's a private pkg higher order function to produce the did, this allows testing the DIDFromPubKey func, however without the Identity provider there's no way to test the error handling in the IdentityFrom* methods, so those will be left untested.